### ISSUE TITLE
db.sqlite

### ISSUE THREAD

**Issue by Kia-Pashang** [tdlib/td#871]
Hellp im using telegram-bot , in the dirctory there is file named **db.sqlite** , this file's size will grow over the time , what things are in this file and how to stop growing size of this file ?

**Comment by levlam** [comment]
You can disable file_database, chat_info_database and message_database using corresponding tdlibParameters fields. Some functionality depending on the databases will be unavailable after that, but the SQLite database will not be used.

**Comment by Kia-Pashang** [comment]
> You can disable file_database, chat_info_database and message_database using corresponding tdlibParameters fields. Some functionality depending on the databases will be unavailable after that, but the SQLite database will not be used.

i used **setOption("use_storage_optimizer","Boolean",true)** does this option work ?

**Comment by levlam** [comment]
@Kia-Pashang This option works, but it isn't related in any way to the database.

**Comment by Kia-Pashang** [comment]
![Capture](https://user-images.githubusercontent.com/26571020/71935716-61748580-315c-11ea-942c-bba5d23ca2c3.JPG)

This config works ?! (tdlib 1.5.0)
or should i try another way ?

**Comment by levlam** [comment]
It should work.

**Comment by Kia-Pashang** [comment]
> It should work.

but it did not work

**Comment by levlam** [comment]
If all databases are disabled, then db.sqlite is not used at all.

**Comment by Kia-Pashang** [comment]
@levlam ill give it another try and ill tell you here

**Comment by levlam** [comment]
@Kia-Pashang Between https://github.com/vysheng/tdbot has no way to change `use_*_database` options. You need to manually replace corresponding parameters in https://github.com/vysheng/tdbot/blob/d976a7010df6c15f385eb504b5289e86bca93834/cliclient.cpp#L266.

**Comment by levlam** [comment]
@egorpugin This is a Linux-specific problem. On Linux you are forced to use exactly the same library version, which is used in TDLib. I would suggest to link your app against SQLCipher 4.4.0 if you need an SQLite database.

**Comment by egorpugin** [comment]
@levlam 
No, I see it on windows. In static and shared builds.
While I can understand static build issue, errors in shared build are a bit tricky.

For shared builds (I did not track the error, so only assumptions) tdlib does not link as dll and built in into app. I have some other 3rdparty lib that is also linked statically (sqlpp11). It seems somehow linker binds wrong/mixed symbols. One from sqlcipher.dll (with statically linked sqlite) and other from sqlite3.dll.

I removed manually normal sqlite3 library during static build linking and everything works.
Need to automate that step...
